docs/contributor-setup.md

```diff
@@ -0,0 +1,42 @@
+# Contributor Setup
+
```

Comment:
unrelated to cuda vs non-cuda, but since we have this page, please mention putting

```
  "rust-analyzer.rustfmt.extraArgs": [
    "+nightly"
   ]
```

for all developers to get our rustfmt to work with the nightly options.